src/transaction/transaction.ts

```diff
@@ -41,23 +43,26 @@ const TxByronWitness = (
 const TxShelleyWitness = (publicKey: Buffer, signature: Buffer): TxWitnessShelley => [publicKey, signature]
 
 const TxAux = (unsignedTxCborHex: UnsignedTxCborHex): _TxAux => {
-  const unsignedTxDecoded:_UnsignedTxDecoded = cbor.decode(unsignedTxCborHex)
-  const parsedTx = parseUnsignedTx(unsignedTxDecoded)
+  const unsignedTxDecoded = cbor.decode(unsignedTxCborHex)
+  if (isUnsignedTxDecoded(unsignedTxDecoded)) {
+    const parsedTx = parseUnsignedTx(unsignedTxDecoded)
 
-  const getId = (): string => {
-    const [txBody] = unsignedTxDecoded
-    const encodedTxBody = encode(txBody)
-    return blake2b(
-      encodedTxBody,
-      32,
-    ).toString('hex')
-  }
+    const getId = (): string => {
+      const [txBody] = unsignedTxDecoded
+      const encodedTxBody = encode(txBody)
+      return blake2b(
+        encodedTxBody,
+        32,
+      ).toString('hex')
+    }
 
-  return {
-    getId,
-    unsignedTxDecoded,
-    ...parsedTx,
+    return {
+      getId,
+      unsignedTxDecoded,
+      ...parsedTx,
+    }
   }
+  throw Error(Errors.InvalidTransactionBody)
```

Comment:
I would like it more if the function tested 
```
if (!isUnsignedTxDecoded(unsignedTxDecoded)) {
   throw Error()
}
```
in the beginning rather than having the whole logic inside if and throwing an exception in the end always (and only the return statement "saving" it).

Reply:
Refactored